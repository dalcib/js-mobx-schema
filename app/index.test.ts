import ClassDecoratorExample from "./index";
import Model from './model';

//const classDecoratorExample = new ClassDecoratorExample()
const example = new ClassDecoratorExample()
const model = new Model(ClassDecoratorExample)

/*const schema = {
  description: 'teste',
  properties: {
    name: { type: 'string', title: 'Name', default: 'fghk' },
    age: { type: 'Numberd', default: 15 },
    ages: {
        maximum: 23,
    },
  date: {
   format: "date",
    }
  }
}*/

describe('non-static decorator', () => {
  it('static', () => {
    expect(ClassDecoratorExample.schema).toBeDefined()
  })
})

describe('Get empty object:', () => {
  it('getEmpty()', () => {
    expect(example.getEmpty()).toMatchSnapshot()
    model.getEmpty()
    expect(model.data).toEqual(example.getEmpty())
  })
})

describe('Get empty model:', () => {
  it('getEmpty()', () => {
    expect(model.data).toMatchSnapshot()
    model.getEmpty()
    expect(model.data).toMatchSnapshot()
  })
})

describe('model', () => {
  it('instanciate', () => {
    expect(model).toMatchSnapshot()
  })
})

describe('static decorator', () => {
  it('static', () => {
    expect(model.schema).toMatchSnapshot()
  })
})
