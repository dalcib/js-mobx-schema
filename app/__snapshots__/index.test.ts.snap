// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get empty model: getEmpty() 1`] = `
ClassDecoratorExample {
  "age": 16,
  "date": undefined,
  "list": undefined,
  "name": "fghk",
}
`;

exports[`Get empty model: getEmpty() 2`] = `
ClassDecoratorExample {
  "age": 16,
  "date": undefined,
  "list": undefined,
  "name": "fghk",
}
`;

exports[`Get empty object: getEmpty() 1`] = `
ClassDecoratorExample {
  "age": 16,
  "date": undefined,
  "list": undefined,
  "name": "fghk",
}
`;

exports[`model instanciate 1`] = `
Model {
  "data": ClassDecoratorExample {
    "age": 16,
    "date": undefined,
    "list": undefined,
    "name": "fghk",
  },
  "schema": Object {
    "description": "teste",
    "properties": Object {
      "age": Object {
        "default": 16,
        "type": "Numberd",
      },
      "ages": Object {
        "maximum": 23,
      },
      "date": Object {
        "format": "date",
      },
      "list": Object {
        "type": "array",
      },
      "name": Object {
        "default": "fghk",
        "title": "Name",
        "type": "string",
      },
    },
  },
}
`;

exports[`static decorator static 1`] = `
Object {
  "description": "teste",
  "properties": Object {
    "age": Object {
      "default": 16,
      "type": "Numberd",
    },
    "ages": Object {
      "maximum": 23,
    },
    "date": Object {
      "format": "date",
    },
    "list": Object {
      "type": "array",
    },
    "name": Object {
      "default": "fghk",
      "title": "Name",
      "type": "string",
    },
  },
}
`;
